### PR TITLE
FIX: fix selection after delete bindings (ISXB-768)

### PR DESCRIPTION
**Pull request by ritamerkl**
### Description

Fixed selection after bindings or actions are deleted ([ISXB-768](https://jira.unity3d.com/browse/ISXB-768)). The selection is set to no selected Item after deleting. This is to match the same behavior like in the old IMGUI editor. 

### Changes made

_Please write down a short description of what changes were made._

### Notes

THIS BRANCH IS BASED ON THE CUT/DELETE BRANCH. Only merge after this one is landed.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Comment by ekcoh** [comment]
I have now merged develop into this branch after landing related cut and paste fix on develop. Resolved merge conflicts and updated this branch. From commits it looks like a lot of work was done on this branch since this was replicated off the original cut-and-paste branch instead of being constructed as a sub-branch off that branch. However diff is smaller now and  allows testing and reviewing this in isolation.

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs [hunk 1]
```diff
@@ -113,16 +113,6 @@ internal struct InputActionsEditorState
             m_CutElements = cutElements;
         }
 
-        private static int AdjustSelection(SerializedObject serializedObject, string propertyName, int index)
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs:116]: This AdjustSelection was dead code introduced in #1850 I decided to remove while merging this with develop.

**Review by graham-huws — APPROVED** [review]
LGTM!

**Review by stefanunity — APPROVED** [review]
Fix works as advertised.

**Pull request merged**